include/treelite/c_api_runtime.h

```diff
@@ -96,75 +41,37 @@ TREELITE_DLL int TreeliteBatchGetDimension(void* handle,
-                                       PredictorHandle* out);
+TREELITE_DLL int TreelitePredictorLoad(
+    const char* library_path, int num_worker_thread, PredictorHandle* out);
 /*!
  * \brief Make predictions on a batch of data rows (synchronously). This
  *        function internally divides the workload among all worker threads.
  * \param handle predictor
- * \param batch a batch of rows (must be of type SparseBatch or DenseBatch)
- * \param batch_sparse whether batch is sparse (1) or dense (0)
+ * \param batch the data matrix containing a batch of rows
  * \param verbose whether to produce extra messages
  * \param pred_margin whether to produce raw margin scores instead of
  *                    transformed probabilities
- * \param out_result resulting output vector; use
- *                   TreelitePredictorQueryResultSize() to allocate sufficient
- *                   space
+ * \param out_result Resulting output vector. This pointer must point to an array of length
+ *                   TreelitePredictorQueryResultSize() and of type
+ *                   TreelitePredictorQueryLeafOutputType().
  * \param out_result_size used to save length of the output vector,
  *                        which is guaranteed to be less than or equal to
  *                        TreelitePredictorQueryResultSize()
  * \return 0 for success, -1 for failure
  */
-TREELITE_DLL int TreelitePredictorPredictBatch(PredictorHandle handle,
-                                               void* batch,
-                                               int batch_sparse,
-                                               int verbose,
-                                               int pred_margin,
-                                               float* out_result,
-                                               size_t* out_result_size);
-
-/*!
- * \brief Make predictions on a single data row (synchronously). The work
- *        will be scheduled to the calling thread.
- * \param handle predictor
- * \param inst single data row
- * \param pred_margin whether to produce raw margin scores instead of
- *                    transformed probabilities
- * \param out_result resulting output vector; use
- *        TreelitePredictorQueryResultSizeSingleInst() to allocate sufficient space
- * \param out_result_size used to save length of the output vector, which is
- *        guaranteed to be at most TreelitePredictorQueryResultSizeSingleInst()
- * \return 0 for success, -1 for failure
- */
-TREELITE_DLL int TreelitePredictorPredictInst(PredictorHandle handle,
-                                              union TreelitePredictorEntry* inst,
-                                              int pred_margin, float* out_result,
-                                              size_t* out_result_size);
-
+TREELITE_DLL int TreelitePredictorPredictBatch(
+    PredictorHandle handle, DMatrixHandle batch, int verbose, int pred_margin,
+    void* out_result, size_t* out_result_size);
```

Comment:
Wouldn't we use "PredictorOutputHandle"(type of out_result) here?

Reply:
The `out_result_size` is actually `LeafOutputType*`, where the type `LeafOutputType` is determined at runtime (by calling `QueryLeafOutputType()`.

Yes, let's use `PredictorOutputHandle` here to clarify the intention of `out_result_size`.

Reply:
Done.